src/hotspot/share/runtime/vmStructs.cpp

```diff
@@ -797,123 +703,8 @@
   static_field(ObjectSynchronizer,             _in_use_list,                                  MonitorList)                           \
   volatile_nonstatic_field(MonitorList,        _head,                                         ObjectMonitor*)                        \
                                                                                                                                      \
-  /*********************/                                                                                                            \
-  /* Matcher (C2 only) */                                                                                                            \
-  /*********************/                                                                                                            \
-                                                                                                                                     \
   unchecked_c2_static_field(Matcher,           _regEncode,                          sizeof(Matcher::_regEncode)) /* NOTE: no type */ \
```

Comment:
I don't see usage in SA of `VMReg::regEncode()` which access this field.

Reply:
This code isn't used but I thought it was somehow needed for stack dumping in one of the Xcomp/SA tests.  But it was really using the presence of the Matcher type to see if isServerCompiler().  So I added the COMPILER2 preprocessor macro and use that instead.